metadata/metadata.go

```diff
@@ -172,28 +174,54 @@ const apiErrorNotFound = "not_found"
 // fetchSingle fetches metadata for the given job, instance, and metric combination.
 // It returns a not-found entry if the fetch is successful but returns no data.
 func (c *Cache) fetchSingle(ctx context.Context, job, instance, metric string) (*cacheEntry, error) {
-	job, instance = escapeLval(job), escapeLval(instance)
-
-	apiResp, err := c.fetch(ctx, "single", url.Values{
-		"match_target": []string{fmt.Sprintf("{job=\"%s\",instance=\"%s\"}", job, instance)},
+	q := url.Values{
 		"metric":       []string{metric},
-	})
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, config.DefaultPrometheusTimeout)
+	defer cancel()
+
+	// TODO: Properly return this error.
+	var retErr error
+	defer fetchTimer.Start(ctx).Stop(&retErr, attribute.String("mode", "single"))
+
+	u := *c.promSimpleURL
+	u.RawQuery = q.Encode()
+
+	req, err := http.NewRequestWithContext(ctx, "GET", u.String(), nil)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "build request")
 	}
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "query Prometheus")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("metadata request HTTP status %s", resp.Status)
+	}
+
+	var apiResp common.SimpleMetadataAPIResponse
+	if err := json.NewDecoder(resp.Body).Decode(&apiResp); err != nil {
+		return nil, errors.Wrap(err, "decode response")
+	}
+
 	now := time.Now()
 
 	if apiResp.ErrorType != "" && apiResp.ErrorType != apiErrorNotFound {
 		return nil, errors.Wrap(errors.New(apiResp.Error), "lookup failed")
 	}
-	if len(apiResp.Data) == 0 {
+	val, ok := apiResp.Data[metric]
+	if !ok {
 		// Cache a not-found entry.
 		return &cacheEntry{
 			lastFetch: now,
 			found:     false,
 		}, nil
 	}
-	d := apiResp.Data[0]
+	d := val[0] // TODO: Explain why the first, etc.
```

Comment:
I recommend we pass a `limit=1` parameter to the API request, so this will make sense. The limit can be explained as an OpenTelemetry requirement, stating that it is a semantic conflict to mix metric point kinds refer to https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#opentelemetry-protocol-data-model

I admit it's not as well spelled out there as it should be, but..